### DISCUSSION TITLE
pyspelling doesn't work

### DISCUSSION THREAD

**Discussion by ingempo** [facelessuser/pyspelling#159] [Q&A]
Try to use in Windows. Aspell 0.50.3 installed. 

./tools/pyspelling/pyspelling.yml
```
matrix:
- name: markdown
  expect_match: false
  apsell:
    mode: en
  pipeline:
  - pyspelling.filters.text:
  sources:
  - '*.txt'
  default_encoding: utf-8
```

Python StackTrace
```
$ python -m pyspelling -vvvv --config ./tools/pyspelling/pyspelling.yml
Using aspell to spellcheck markdown
Running Task: markdown...

> Processing: text.txt

b'cheking'
Command: ['aspell', 'list', '--encoding', 'utf-8', '--mode', 'none']
Context: text.txt
ERROR: text.txt -- [WinError 2] The system cannot find the file specified

!!!Spelling check failed!!!
```

Is there something I am doing wrong?

**Comment by facelessuser** [comment]
I've only ever tested Pyspelling in Windows via an MSYS2/MinGW environment: https://facelessuser.github.io/pyspelling/#usage-in-windows.

I suppose using it in WSL would also work. I believe 0.50.3 is likely too old a version of Aspell, but I am not sure. On mac, I'm using this version:

```
@(#) International Ispell Version 3.1.20 (but really Aspell 0.60.8)
```

**Reply by ingempo** [reply]
Me too ! 
```
$ aspell --version
@(#) International Ispell Version 3.1.20 (but really Aspell 0.60.8)
```
but:

```
$ python -m pyspelling -vvvv --config ./tools/pyspelling/pyspelling.yml
Using aspell to spellcheck markdown
Running Task: markdown...

> Processing: text.txt

b'cheking'
Command: ['aspell', 'list', '--encoding', 'utf-8', '--mode', 'none']
Context: text.txt
ERROR: text.txt -- Runtime Error: Error: Unknown mode: "none".

!!!Spelling check failed!!!
```
However:
```
$ aspell -c text.txt 
*cheking*1) checking                             5) chocking
2) cheeking                             6) chucking
3) choking                              7) shaking
4) chewing
i) Ignore                               I) Ignore all
r) Replace                              R) Replace all
a) Add                                  l) Add Lower
b) Abort                                x) Exit
? x
```

**Reply by facelessuser** [reply]
I need more info. Are you running WSL or MSYS2, how are you running things as I have no idea. Do you need to do the `--binary` option to point pyspelling at the aspell binary?

**Reply by ingempo** [reply]
I'm in MSYS2/MINGW64.

Well I undertstand PySpelling is a wrapper for Aspell. I don't need `--binary` switch. In fact, the error comes directly from Aspell, but PySpelling puts by default the switch `--mode` in the `command` even if I delete it from `pyspelling.yml`

**Comment by facelessuser** [comment]
See, I'm confused as you are telling me that that you are using `Try to use in Windows. Aspell 0.50.3 installed.` in the opening page, but then showing me you are using `Aspell 0.60.8`? Which is it? Which executable is being found by Pyspelling?

If mode is not defined, we set it to `none`. We do this to prevent Aspell from using filters unless we want them to.

```python
        if 'mode' not in options:
            options['mode'] = 'none'
```

I have absolutely no issues with my version (0.60.8) recognizing `none` mode.  It is even documented in Aspell's documentation: http://aspell.net/0.61/man-html/Notes-on-Various-Filters-and-Filter-Modes.html.

So, if Aspell does not know what to do with `none` for `mode`, then you are probably not actually using `0.60.8`. Pyspelling is likely accessing the `0.50.3` version you referenced in the OP.

**Reply by ingempo** [reply]
Could please do:
```
aspell dump modes
```
and give me the output?

For `filters` and `dicts` I get an output but not with `modes`. I don't undestand why...

**Reply by facelessuser** [reply]
You are most likely using an incompatible version, mine works fine:

```
➜  pyspelling git:(master) ✗ aspell dump modes
nroff          mode for checking Nroff documents
perl           mode for checking Perl comments and string literals
email          mode for skipping quoted text in email messages
tex            mode for checking TeX/LaTeX documents
markdown       mode for checking Markdown/CommonMark documents
ccpp           mode for checking C++ comments and string literals
sgml           mode for checking generic SGML/XML documents
none           mode to disable all filters
texinfo        mode for checking Texinfo documents
html           mode for checking HTML documents
comment        mode to check any lines starting with a #
url            mode to skip URL like constructs (default mode)
```
